content/structure.md

```diff
@@ -0,0 +1,390 @@
+
+
+<h3 id="eslint-commit-hook">Setting up a commit hook for linting</h3>
+
+XXX I think we just don't worry about these for now?
+
+<h3 id="eslint-ci">Running ESLint in your CI environment</h3>
+
+XXX
+
+<h2 id="meteor-features">Meteor features</h2>
+
+The section above talked about JavaScript code in general - you can easily apply it in any JavaScript application, not just with Meteor apps. However, there are some style questions that are Meteor-specific, in particular how to name and structure all of the different components of your app.
+
+<h3 id="collections">Collections</h3>
+
+Collections should be named as a plural noun, in PascalCase. The name of the collection in the database (the first argument to the collection constructor) should be the same as the name of the JavaScript symbol.
+
+```js
+// Defining a collection
+Lists = new Mongo.Collection('Lists');
+```
+
+Fields in the database should be camelCased just like your JavaScript variable names.
+
+```js
+// Inserting a document with camelCased field names
+Widgets.insert({
+  myFieldName: 'Hello, world!',
+  otherFieldName: 'Goodbye.'
+});
+```
+
+<h3 id="methods-and-publications">Methods and publications</h3>
+
+Method and publication names should be camelCased, and namespaced to the collection or module:
+
+```js
+// in imports/api/todos/methods.js
+updateText = new ValidatedMethod({
+  name: 'Todos.methods.updateText',
+  // ...
+});
+```
+
+// XXX: should we just make the method name `Todos.updateText`? --- it make sense to qualify it with `.method` when it was also the JS name, but that's no longer the case...
+
+Note that this code sample uses the [ValidatedMethod package recommended in the Methods article](methods.html#validated-method). If you aren't using that package, you can use the name as the property passed to `Meteor.methods`.
+
+Here's how this naming convention looks when applied to a publication:
+
+```js
+// Naming a publication
+Meteor.publish('Lists.public', function() {
+  // ...
+});
+```
+
+<h3 id="files-and-exports">Files, exports, and packages</h3>
+
+You should use the ES2015 `import` and `export` features to manage your code. This will let you better understand the dependencies between different parts of your code, and it will be easy to know where to look if you need to read the source code of a dependency.
+
+Each file in your app should represent one logical module. Avoid having catch-all utility modules that export a variety of unrelated functions and symbols. Often, this can mean that it's good to have one class, UI component, or collection per file, but there are cases where it is OK to make an exception, for example if you have a UI component with a small sub-component that isn't used outside of that file.
+
+When a file represents a single class or UI component, the file should be named the same as the thing it defines, with the same capitalization. So if you have a file that exports a class:
+
+```js
+export class ClickCounter { ... }
```

Comment:
This won't actually work - for the below import to work this would have to be an `export default`.

Personally, I don't know if it's better to have a default export or not, what do you think we should recommend here @benjamn?